app/controllers/admin/topics_controller.rb

```diff
@@ -182,36 +182,40 @@ def update_topic
   # Assigns a discussion to another agent
   def assign_agent
 
-    @count = 0
-    #handle array of topics
-    params[:topic_ids].each do |id|
-
-      @topic = Topic.where(id: id).first
-
-      @minutes = 0
-#      logger.info("inside loop")
-#      logger.info(params[:assigned_user_id])
-      unless params[:assigned_user_id].blank?
-
+    #@count = 0
+    assigned_user = User.find(params[:assigned_user_id])
+    @topics = Topic.where(id: params[:topic_ids])
+    bulk_post_attributes = []
+    unless params[:assigned_user_id].blank?
+      #handle array of topics
+      @topics.each do |topic|
+
+        @minutes = 0
+        #logger.info("inside loop")
+        #logger.info(params[:assigned_user_id])
+        
         # if message was unassigned previously, use the new assignee
         # this is to give note attribution below
-        previous_assigned_id = @topic.assigned_user_id? ? @topic.assigned_user_id : params[:assigned_user_id]
-        assigned_user = User.find(params[:assigned_user_id])
-        @topic.assign(previous_assigned_id, assigned_user.id)
+        previous_assigned_id = topic.assigned_user_id || params[:assigned_user_id]
+        bulk_post_attributes << {body: I18n.t(:assigned_message, assigned_to: assigned_user.name), kind: 'note', user_id: previous_assigned_id, topic_id: topic.id}
+        #topic.assign(previous_assigned_id, assigned_user.id)
 
         # Create internal note
-#        @topic.posts.create(user_id: previous_assigned_id, body: "Discussion has been transferred to #{assigned_user.name}.", kind: "note")
+        # @topic.posts.create(user_id: previous_assigned_id, body: "Discussion has been transferred to #{assigned_user.name}.", kind: "note")
 
         # Calls to GA
-        @tracker.event(category: "Agent: #{current_user.name}", action: "Assigned to #{assigned_user.name.titleize}", label: @topic.to_param, value: @minutes)
+        @tracker.event(category: "Agent: #{current_user.name}", action: "Assigned to #{assigned_user.name.titleize}", label: topic.to_param, value: @minutes)
 
+        #@count = @count + 1
       end
-      @count = @count + 1
-    end
+    end 
+    
+    @topics.bulk_assign(bulk_post_attributes, assigned_user.id) if bulk_post_attributes.present?
 
     if params[:topic_ids].count > 1
       get_tickets
     else
+      @topic = @topics.first
```

Comment:
Subtle change in behaviour here: previous code would be equivalent to `@topic = @topics.last` I think? Or - at least - it would have been the topic for the last ID in `params[:topic_ids]` (where the sort order is unspecified?). Not sure whether this is important anyway as don't know what `@topic` is used for.